### ISSUE TITLE
Time of Day Not Followed

### ISSUE THREAD

**Issue by zgosalvez** [JohnCoates/Aerial#1136]
#### General troubleshooting tips

Before logging an issue please check that:
- [x] You have the latest version installed (There may be a beta version that fixes your issue), see here for the latest releases and bug fixes: https://github.com/JohnCoates/Aerial/releases
- [x] Your issue isn't already mentioned in our [issues](https://github.com/JohnCoates/Aerial/issues). You may find a workaround there or a similar request already made.
- [x] Your problem isn't mentioned in the [troubleshooting page](https://github.com/JohnCoates/Aerial/blob/master/Documentation/Troubleshooting.md).

If none of this fixes your issue, tell us about the problem you are experiencing or the feature you'd like to request.

#### Required information
In order to help us sort your issue, we ask that you provide the following information:
- [x] Mac model: MacBook Pro (13-inch, 2020, Four Thunderbolt 3 ports)
- [x] macOS version: 11.0.1 (20B50)
- [x] Monitor setup: Dual spanned

If appropriate, please enable `Debug mode` and `Log to disk` in `Advanced` tab and replicate your bug, then attach the `AerialLog.txt` file (You can access this file through the Advanced tab).

```
2020-12-02 09:27:03.081 : avInit .saver (0.0, 0.0, 1440.0, 900.0) p: false o: false
2020-12-02 09:27:03.083 : Time Management initialized
2020-12-02 09:27:03.084 : Starting Location initialization
2020-12-02 09:27:03.084 : Location services enabled
2020-12-02 09:27:03.084 : <AerialView: 0x7fcb6470ce20> AerialView setup init (V2.2.1) preview: false
2020-12-02 09:27:03.085 : Running Version 11.0.1 (Build 20B50)
2020-12-02 09:27:03.085 : Display Detection initialized
2020-12-02 09:27:03.086 : ***Display Detection***
2020-12-02 09:27:03.086 : 2 display(s) detected
2020-12-02 09:27:03.086 : 1cm = 50.393701544239036 points
2020-12-02 09:27:03.087 : npass: dict [__C.NSDeviceDescriptionKey(_rawValue: NSDeviceResolution): NSSize: {144, 144}, __C.NSDeviceDescriptionKey(_rawValue: NSDeviceBitsPerSample): 8, __C.NSDeviceDescriptionKey(_rawValue: NSDeviceColorSpaceName): NSCalibratedRGBColorSpace, __C.NSDeviceDescriptionKey(_rawValue: NSScreenNumber): 69734208, __C.NSDeviceDescriptionKey(_rawValue: NSDeviceIsScreen): YES, __C.NSDeviceDescriptionKey(_rawValue: NSDeviceSize): NSSize: {1440, 900}]
2020-12-02 09:27:03.087 :        bottomLeftFrame (0.0, 0.0, 1440.0, 900.0)
2020-12-02 09:27:03.087 : npass: dict [__C.NSDeviceDescriptionKey(_rawValue: NSDeviceResolution): NSSize: {72, 72}, __C.NSDeviceDescriptionKey(_rawValue: NSDeviceBitsPerSample): 8, __C.NSDeviceDescriptionKey(_rawValue: NSDeviceColorSpaceName): NSCalibratedRGBColorSpace, __C.NSDeviceDescriptionKey(_rawValue: NSScreenNumber): 441001153, __C.NSDeviceDescriptionKey(_rawValue: NSDeviceIsScreen): YES, __C.NSDeviceDescriptionKey(_rawValue: NSDeviceSize): NSSize: {1920, 1080}]
2020-12-02 09:27:03.087 :        bottomLeftFrame (-1920.0, 496.0, 1920.0, 1080.0)
2020-12-02 09:27:03.088 : src orig : (0.0, 0.0)
2020-12-02 09:27:03.088 : left 1.0 below 0.0
2020-12-02 09:27:03.088 : src orig : (-1920.0, 496.0)
2020-12-02 09:27:03.088 : left 0.0 below 0.0
2020-12-02 09:27:03.089 : [id=69734208, width=1440, height=900, bottomLeftFrame=(0.0, 0.0, 1440.0, 900.0), topRightCorner=(1440.0, 900.0), isMain=true, backingScaleFactor=2.0]
2020-12-02 09:27:03.089 : [id=441001153, width=1920, height=1080, bottomLeftFrame=(-1920.0, 496.0, 1920.0, 1080.0), topRightCorner=(0.0, 1576.0), isMain=false, backingScaleFactor=1.0]
2020-12-02 09:27:03.089 : (-1920.0, 0.0, 3360.0, 1576.0)
2020-12-02 09:27:03.089 : ***Display Detection Done***
2020-12-02 09:27:03.090 : Real screen count : 2
2020-12-02 09:27:03.090 : Using : Optional([id=69734208, width=1440, height=900, bottomLeftFrame=(0.0, 0.0, 1440.0, 900.0), topRightCorner=(1440.0, 900.0), isMain=true, backingScaleFactor=2.0])
2020-12-02 09:27:03.090 : <AerialView: 0x7fcb6470ce20> setting up player layer with bounds/frame: (0.0, 0.0, 1440.0, 900.0) / (0.0, 0.0, 1440.0, 900.0)
2020-12-02 09:27:03.092 : allDisplays
2020-12-02 09:27:03.092 : mainOnly
2020-12-02 09:27:03.093 : mainOnly
2020-12-02 09:27:03.093 : Using dot workaround for video driver corruption
2020-12-02 09:27:03.093 : tvOS 13 is enabled, cached and up to date
2020-12-02 09:27:03.094 : tvOS 12 is enabled, cached and up to date
2020-12-02 09:27:03.094 : tvOS 11 is enabled, cached and up to date
2020-12-02 09:27:03.094 : tvOS 10 is enabled, cached and up to date
2020-12-02 09:27:03.361 : <AerialView: 0x7fcb6470ce20> backing change 2.0 isDisabled: false frame: (0.0, 0.0, 1440.0, 900.0) preview: false
2020-12-02 09:27:03.362 : LMauth status change : 3
2020-12-02 09:27:03.362 : Location coordinate : CLLocationCoordinate2D(latitude: [redacted], longitude: [redacted])
2020-12-02 09:27:03.363 : Location found [redacted] [redacted]
2020-12-02 09:27:03.365 : Refreshing video list
2020-12-02 09:27:03.365 : Poi Strings Provider initialized
2020-12-02 09:27:03.365 : Preferred languages : ["en-PH", "fil-PH"]
2020-12-02 09:27:03.366 : Best matched language : en
2020-12-02 09:27:03.366 : Defaulting to bundle
2020-12-02 09:27:03.366 : path : /Users/[redacted]/Library/Screen Savers/Aerial.saver/Contents/Resources/en.json
2020-12-02 09:27:03.366 : Community JSON : 99 entries
2020-12-02 09:27:03.367 : source : tvOS 13 contains 90 new videos (total 90)
2020-12-02 09:27:03.367 : source : tvOS 12 contains 0 new videos (total 90)
2020-12-02 09:27:03.367 : source : tvOS 11 contains 0 new videos (total 90)
2020-12-02 09:27:03.368 : source : tvOS 10 contains 9 new videos (total 99)
2020-12-02 09:27:03.368 : <AerialView: 0x7fcb6470ce20> pnv
2020-12-02 09:27:03.368 : <AerialView: 0x7fcb6470ce20> Setting player for all player layers in [<AerialView: 0x7fcb6470ce20>]
2020-12-02 09:27:03.369 : remaining in playlist : 0
2020-12-02 09:27:03.369 : generate playlist
2020-12-02 09:27:03.369 : playlist count: 91
2020-12-02 09:27:03.369 : Final count : 6
2020-12-02 09:27:03.370 : initvideocache
2020-12-02 09:27:03.370 : cached video file with length: 196041941
2020-12-02 09:27:03.370 : AssetLoaderDelegate deinit
2020-12-02 09:27:03.371 : <AerialView: 0x7fcb6470ce20> playing video (OFFLINE MODE) : file:///Users/[redacted]/Library/Containers/com.apple.ScreenSaver.Engine.legacyScreenSaver/Data/Library/Application%20Support/Aerial/Cache/comp_GMT306_139NC_139J_3066_CALI_TO_VEGAS_v08_HDR_PS_20180824_HDR_4K_HEVC.mov
2020-12-02 09:27:03.371 : <AerialView: 0x7fcb6470ce20> observing current item <AVPlayerItem: 0x600000fb1ea0, asset = <AVURLAsset: 0x600000df93c0, URL = file:///Users/[redacted]/Library/Containers/com.apple.ScreenSaver.Engine.legacyScreenSaver/Data/Library/Application%20Support/Aerial/Cache/comp_GMT306_139NC_139J_3066_CALI_TO_VEGAS_v08_HDR_PS_20180824_HDR_4K_HEVC.mov>>
2020-12-02 09:27:03.371 : <AerialView: 0x7fcb6470ce20> observeValue Optional("readyForDisplay") false
2020-12-02 09:27:03.375 : avInit .saver (-1920.0, 496.0, 1920.0, 1080.0) p: false o: false
2020-12-02 09:27:03.375 : <AerialView: 0x7fcb64414780> AerialView setup init (V2.2.1) preview: false
2020-12-02 09:27:03.376 : Running Version 11.0.1 (Build 20B50)
2020-12-02 09:27:03.376 : Real screen count : 2
2020-12-02 09:27:03.377 : Using : Optional([id=441001153, width=1920, height=1080, bottomLeftFrame=(-1920.0, 496.0, 1920.0, 1080.0), topRightCorner=(0.0, 1576.0), isMain=false, backingScaleFactor=1.0])
2020-12-02 09:27:03.377 : <AerialView: 0x7fcb64414780> setting up player layer with bounds/frame: (0.0, 0.0, 1920.0, 1080.0) / (0.0, 0.0, 1920.0, 1080.0)
2020-12-02 09:27:03.377 : allDisplays
2020-12-02 09:27:03.378 : mainOnly
2020-12-02 09:27:03.378 : mainOnly
2020-12-02 09:27:03.379 : Using dot workaround for video driver corruption
2020-12-02 09:27:03.379 : <AerialView: 0x7fcb64414780> backing change 1.0 isDisabled: false frame: (0.0, 0.0, 1920.0, 1080.0) preview: false
2020-12-02 09:27:03.590 : <AerialView: 0x7fcb6470ce20> observeValue Optional("readyForDisplay") true
2020-12-02 09:27:03.591 : start playback: (0.0, 0.0, 1440.0, 900.0) (0.0, 0.0, 1440.0, 900.0)
2020-12-02 09:27:03.591 : <AerialView: 0x7fcb6470ce20> startAnimation frame (0.0, 0.0, 1440.0, 900.0) bounds (0.0, 0.0, 1440.0, 900.0)
2020-12-02 09:27:03.876 : <AerialView: 0x7fcb64414780> startAnimation frame (0.0, 0.0, 1920.0, 1080.0) bounds (0.0, 0.0, 1920.0, 1080.0)
2020-12-02 09:28:38.235 : Location using cached data
2020-12-02 09:28:52.348 : Location using cached data
```

#### Description of issue / Feature request

The Time setting is "Use my location". If I click Test, it shows a success message that Aerial can access my location. However, the video being played doesn't match the current time. For example, it's already 9 AM but it's playing nighttime videos.

**Comment by glouel** [comment]
Hey @zgosalvez 

Thanks for the report, I will have to put up a beta with better logging as I'm not quite sure what's happening here. Cali to vegas is indeed a night video.

A few things if you could check : 
- Can you go into the time tab and tell me what you have in your bar at the bottom ? Here's mine to give you an example, at 9:27, I'm still in "sunrise" based on my settings : 

<img width="1048" alt="Capture d’écran 2020-12-02 à 14 41 55" src="https://user-images.githubusercontent.com/37544189/100880032-88daec00-34ac-11eb-9794-aec408a5abba.png">

I'm asking because in some circumstances (you don't have any sunrise video cached), Aerial will fallback to something else. 

- Second, what do you have for Cache settings ? How large is your cache limit, and how much do you have downloaded ? 

Based on that I'll put a new beta up today to see if I can narrow it down a bit. Thanks

**Comment by glouel** [comment]
@zgosalvez please try 2.2.2beta1, can you launch it once in full screen, and show me the log ? Thanks !

https://github.com/JohnCoates/Aerial/releases/download/v2.2.2beta1/Aerial.saver.zip

**Comment by zgosalvez** [comment]
@glouel, thank you for debugging this with me.

* Time settings
<img width="988" alt="Screen Shot 2020-12-03 at 08 51 57" src="https://user-images.githubusercontent.com/1798166/100949384-1f5aec00-3545-11eb-99d3-de53f0d7f77a.png">

* Cache settings
<img width="1109" alt="Screen Shot 2020-12-03 at 08 57 34" src="https://user-images.githubusercontent.com/1798166/100949629-a27c4200-3545-11eb-996f-78012c374b7b.png">

* 2.2.2beta1 log
```
2020-12-03 08:52:25.471 : avInit .saver (0.0, 0.0, 1440.0, 900.0) p: false o: false
2020-12-03 08:52:25.472 : Time Management initialized
2020-12-03 08:52:25.472 : Starting Location initialization
2020-12-03 08:52:25.473 : Location services enabled
2020-12-03 08:52:25.473 : <AerialView: 0x7fe394c09cc0> AerialView setup init (V2.2.2beta1) preview: false
2020-12-03 08:52:25.473 : Running Version 11.0.1 (Build 20B50)
2020-12-03 08:52:25.473 : Display Detection initialized
2020-12-03 08:52:25.474 : ***Display Detection***
2020-12-03 08:52:25.474 : 1 display(s) detected
2020-12-03 08:52:25.474 : 1cm = 50.393701544239036 points
2020-12-03 08:52:25.474 : npass: dict [__C.NSDeviceDescriptionKey(_rawValue: NSScreenNumber): 69734208, __C.NSDeviceDescriptionKey(_rawValue: NSDeviceResolution): NSSize: {144, 144}, __C.NSDeviceDescriptionKey(_rawValue: NSDeviceSize): NSSize: {1440, 900}, __C.NSDeviceDescriptionKey(_rawValue: NSDeviceBitsPerSample): 8, __C.NSDeviceDescriptionKey(_rawValue: NSDeviceIsScreen): YES, __C.NSDeviceDescriptionKey(_rawValue: NSDeviceColorSpaceName): NSCalibratedRGBColorSpace]
2020-12-03 08:52:25.475 :        bottomLeftFrame (0.0, 0.0, 1440.0, 900.0)
2020-12-03 08:52:25.475 : src orig : (0.0, 0.0)
2020-12-03 08:52:25.475 : left 0.0 below 0.0
2020-12-03 08:52:25.475 : [id=69734208, width=1440, height=900, bottomLeftFrame=(0.0, 0.0, 1440.0, 900.0), topRightCorner=(1440.0, 900.0), isMain=true, backingScaleFactor=2.0]
2020-12-03 08:52:25.475 : (0.0, 0.0, 1440.0, 900.0)
2020-12-03 08:52:25.476 : ***Display Detection Done***
2020-12-03 08:52:25.476 : Real screen count : 1
2020-12-03 08:52:25.476 : Using : Optional([id=69734208, width=1440, height=900, bottomLeftFrame=(0.0, 0.0, 1440.0, 900.0), topRightCorner=(1440.0, 900.0), isMain=true, backingScaleFactor=2.0])
2020-12-03 08:52:25.476 : <AerialView: 0x7fe394c09cc0> setting up player layer with bounds/frame: (0.0, 0.0, 1440.0, 900.0) / (0.0, 0.0, 1440.0, 900.0)
2020-12-03 08:52:25.478 : allDisplays
2020-12-03 08:52:25.478 : mainOnly
2020-12-03 08:52:25.478 : mainOnly
2020-12-03 08:52:25.479 : Using dot workaround for video driver corruption
2020-12-03 08:52:25.479 : tvOS 13 is enabled, cached and up to date
2020-12-03 08:52:25.479 : tvOS 12 is enabled, cached and up to date
2020-12-03 08:52:25.479 : tvOS 11 is enabled, cached and up to date
2020-12-03 08:52:25.480 : tvOS 10 is enabled, cached and up to date
2020-12-03 08:52:25.779 : <AerialView: 0x7fe394c09cc0> backing change 2.0 isDisabled: false frame: (0.0, 0.0, 1440.0, 900.0) preview: false
2020-12-03 08:52:25.780 : LMauth status change : 3
2020-12-03 08:52:25.780 : Location coordinate : CLLocationCoordinate2D(latitude: [redacted], longitude: [redacted])
2020-12-03 08:52:25.781 : Location found [redacted] [redacted]
2020-12-03 08:52:25.783 : Refreshing video list
2020-12-03 08:52:25.783 : Poi Strings Provider initialized
2020-12-03 08:52:25.783 : Preferred languages : ["en-PH", "fil-PH"]
2020-12-03 08:52:25.783 : Best matched language : en
2020-12-03 08:52:25.783 : Defaulting to bundle
2020-12-03 08:52:25.784 : path : /Users/[redacted]/Library/Screen Savers/Aerial.saver/Contents/Resources/en.json
2020-12-03 08:52:25.784 : Community JSON : 99 entries
2020-12-03 08:52:25.784 : source : tvOS 13 contains 90 new videos (total 90)
2020-12-03 08:52:25.784 : source : tvOS 12 contains 0 new videos (total 90)
2020-12-03 08:52:25.785 : source : tvOS 11 contains 0 new videos (total 90)
2020-12-03 08:52:25.785 : source : tvOS 10 contains 9 new videos (total 99)
2020-12-03 08:52:25.785 : <AerialView: 0x7fe394c09cc0> pnv
2020-12-03 08:52:25.785 : <AerialView: 0x7fe394c09cc0> Setting player for all player layers in [<AerialView: 0x7fe394c09cc0>]
2020-12-03 08:52:25.785 : Dark Mode override
2020-12-03 08:52:25.786 : remaining in playlist : 0
2020-12-03 08:52:25.786 : generate playlist
2020-12-03 08:52:25.786 : Playlist raw count: 91 raw cached count 25 isRestricted: true restrictedTo: night
2020-12-03 08:52:25.786 : Final count : 6
2020-12-03 08:52:25.787 : initvideocache
2020-12-03 08:52:25.787 : cached video file with length: 369968186
2020-12-03 08:52:25.787 : AssetLoaderDelegate deinit
2020-12-03 08:52:25.787 : <AerialView: 0x7fe394c09cc0> playing video (OFFLINE MODE) : file:///Users/[redacted]/Library/Containers/com.apple.ScreenSaver.Engine.legacyScreenSaver/Data/Library/Application%20Support/Aerial/Cache/comp_GMT060_117NC_363D_1034_AUSTRALIA_v35_HDR_PS_FINAL_20180831_HDR_4K_HEVC.mov
2020-12-03 08:52:25.788 : <AerialView: 0x7fe394c09cc0> observing current item <AVPlayerItem: 0x600003945d90, asset = <AVURLAsset: 0x600003b75900, URL = file:///Users/[redacted]/Library/Containers/com.apple.ScreenSaver.Engine.legacyScreenSaver/Data/Library/Application%20Support/Aerial/Cache/comp_GMT060_117NC_363D_1034_AUSTRALIA_v35_HDR_PS_FINAL_20180831_HDR_4K_HEVC.mov>>
2020-12-03 08:52:25.788 : <AerialView: 0x7fe394c09cc0> observeValue Optional("readyForDisplay") false
2020-12-03 08:52:25.978 : <AerialView: 0x7fe394c09cc0> startAnimation frame (0.0, 0.0, 1440.0, 900.0) bounds (0.0, 0.0, 1440.0, 900.0)
2020-12-03 08:52:26.004 : <AerialView: 0x7fe394c09cc0> observeValue Optional("readyForDisplay") true
2020-12-03 08:52:26.005 : start playback: (0.0, 0.0, 1440.0, 900.0) (0.0, 0.0, 1440.0, 900.0)
```

**Comment by zgosalvez** [comment]
@glouel, I think the "Show only night videos in Dark Mode". I tried turning it off and so far the screensaver is playing day time videos. Note that System Preferences > General > Appearance is set to Auto. I noticed this same behavior before with SourceTree's theme when it relies on the system appearance. SourceTree falsely detects dark mode when the system apperance is set to Auto.

**Comment by glouel** [comment]
This is super interesting @zgosalvez and might solve another issue that was reported to me, I will look into how my code works when Appearance is set to Auto and see if there's something here. Thanks a whole lot I'll get back to you.

**Comment by glouel** [comment]
@zgosalvez this should be fixed in 2.2.3 now ! Thanks again for the report!

**Comment by zgosalvez** [comment]
@glouel, was my pleasure. Thanks for the quick fix!
